Yafc.Parser/Data/FactorioDataDeserializer_Context.cs

```diff
@@ -273,14 +277,17 @@ private void CalculateMaps(bool netProduction) {
                     }
 
                     break;
+
```

Comment:
Do we really need these double newlines around the `break` (well, above and below?) I would prefer to keep it at just one, or none.

Reply:
My idea was that we definitely don't want to miss a `break` statement in a `switch`, even at 3AM.  So I put a blank line at least below it. The lines above it are usually due to a flow statement like `if` or `while` ending, or due to the cases around this case having blank lines around `break`s for the same reason.

Reply:
I'm not sure i can agree with a "by default" newline around `if` and `while` statements. I agree many of them in this pr are good, so it's a good rule of thumb, but sometimes the line right above an `if` or `while` does belong together.

Reply:
I picked up this rule from my job. My understanding of it is that you don't want to mistake where `if` or `while` start and end, so you give them some space. I wrote down my understanding of that rule in the [code style](https://github.com/shpaass/yafc-ce/blob/master/Docs/CodeStyle.md#blank-lines).

Reply:
I'm guessing the rule for breaks in switches is related to C, C++, and/or Java. In C#, the compiler requires a break (or other unconditional jump) at the end of any `case` region. This common error is not allowed:
``` c
case 1:
    // something
    // Oops! no break
case 2:
```
If that's the desired behavior, you have to explicitly say
``` c
case 1:
    // something
    goto case 2;
case 2:
```

Which is a long way of saying that I also don't think there needs to be a blank line between `break` (or `return`, `throw`, etc.) and `case`.

Reply:
Actually no break is allowed on empty case switched, but not if it contains anything.

Reply:
Wow. Indeed there is [CS0163](https://learn.microsoft.com/en-us/dotnet/csharp/language-reference/compiler-messages/cs0163). Then I agree that we don't need blank lines between `break` an `case`.

Reply:
I've removed what I found in https://github.com/shpaass/yafc-ce/pull/270/commits/873bd21f5717b9b59e75c7a47600809f63f67382

Please let me know if I missed something

Reply:
I checked at least everything in Yafc.UI and nothing missed there. Well there was one return statement, but in that case i think it makes sense to let it stand out (Yafc.UI\ImGui\ImGuiTextInputHelper.cs:278) so i'm for keeping that in.